music.asm

```diff
@@ -289,10 +289,14 @@ Overworld_MosaicDarkWorldChecks:
 ; 
 ; On entry, A=16bit XY=8bit, A & X safe to mod, Y unknown
 Underworld_DoorDown_Entry:
-    LDA.l DRMode : TAX : LDA.w #0 : CPX #0 : BNE .done
-
-.vanilla
-    LDX #$14 : LDA $A0  ; thing we wrote over - prepare Sanc music track, load current room ID
+    LDA.l DRMode : TAX : LDA $A0 : CPX #0 : BNE .done
+
+.vanilla ; thing we wrote over
+    LDX #$14 ;: LDA $A0
+    CMP.w #$0012 : BEQ .done
+    
+    LDX.b #$10 ; value for Hyrule Castle music
+    CMP.w #$0002 : BEQ .done
```

Comment:
this is odd, the LDX, CMP, and BEQ here all seem to be no-ops unless the flags and X value is important because everything simply just goes to the done label no matter what

Reply:
Yeah, when this exits, the value of X is the music track that is loaded, and when DR isn't enabled, A is set to the room ID of the room being loaded